### ISSUE TITLE
NPM State doesn't install requested version

### ISSUE THREAD

**Issue by syphernl** [saltstack/salt#21064]
Installing a package with a particular version doesn't appear to work. It is installing the latest version.

Additionally, every highstate it is marked as changed because the current version doesn't match the requested one.

Example SLS:
```
pm2@0.9.2:
  npm.installed:
    - require:
      - pkg: nodejs
```

Each run shows:
```
          ID: pm2@0.9.2
    Function: npm.installed
      Result: True
     Comment: Package(s) 'pm2@0.9.2' successfully installed
     Started: 09:11:32.958451
    Duration: 6739.848 ms
     Changes:   
              ----------
              new:
                  - pm2@0.9.2
              old:
```

It appears that the version is not being used at all:
```
[INFO    ] Running state [pm2@0.9.2] at time 09:11:32.958451
[INFO    ] Executing state npm.installed for pm2@0.9.2
[INFO    ] Executing command 'npm --version' in directory '/home/vagrant'
[INFO    ] Executing command 'npm list --silent --json --global' in directory '/home/vagrant'
[INFO    ] Executing command 'npm --version' in directory '/home/vagrant'
[INFO    ] Executing command 'npm install --silent --json --global "pm2"' in directory '/home/vagrant'
```

**Comment by rallytime** [comment]
@syphernl What version of salt are your running? Can you please post the output of `salt --versions-report`? I just tested this the other day in 2014.7.2 and it was working for me. See #18647.

**Comment by syphernl** [comment]
I'm on the current stable release (as is available in the Ubuntu repo) on Ubuntu 14.04:
```
           Salt: 2014.7.1
         Python: 2.7.6 (default, Mar 22 2014, 22:59:56)
         Jinja2: 2.7.2
       M2Crypto: 0.21.1
 msgpack-python: 0.3.0
   msgpack-pure: Not Installed
       pycrypto: 2.6.1
        libnacl: Not Installed
         PyYAML: 3.10
          ioflo: Not Installed
          PyZMQ: 14.0.1
           RAET: Not Installed
            ZMQ: 4.0.4
           Mako: 0.9.1
```

**Comment by rallytime** [comment]
@syphernl Yes, this is a duplicate of the issue linked above and has been fixed in 2014.7.2. Hopefully those packages will be available soon in the Ubuntu repo (I thought they were already, but after checking with `apt-get -u update` I don't see them listed). 

I just gave this another try on 2014.7.2, just to confirm:
```
[DEBUG   ] Rendered data from file: /var/cache/salt/minion/files/base/npm-test.sls:
coffee-script@1.0.1:
  npm.installed: []

[DEBUG   ] Results of YAML rendering:
OrderedDict([('coffee-script@1.0.1', OrderedDict([('npm.installed', [])]))])
[INFO    ] Running state [coffee-script@1.0.1] at time 00:07:52.032889
[INFO    ] Executing state npm.installed for coffee-script@1.0.1
```
Since this is a duplicate and it has already been fixed, I'm going to close this issue. Thanks again for the report, and hopefully @joehealy will be able to get those new Ubuntu packages posted soon.
